src/eos_gates.js

```diff
@@ -150,6 +151,38 @@ const Application = new Lang.Class({
     },
 });
 
+function spawnProcess(argv=[]) {
+    return Gio.Subprocess.new(argv, Gio.SubprocessFlags.NONE);
+}
+
+function getAppStoreAppId(remote, appId) {
+    let installation = Flatpak.Installation.new_system(null);
+    let flatpakRemote = null;
+
+    let defaultBranch = null;
+    try {
+        flatpakRemote = installation.get_remote_by_name(remote, null);
+    } catch (e) {
+        logError(e, 'Could not find flatpak remote %s: %s'.format(remote));
+    }
+
+    // Get the default branch now to construct the full unique ID GNOME Software expects.
+    if (flatpakRemote) {
+        defaultBranch = flatpakRemote.get_default_branch()
+        if (defaultBranch)
+            return 'system/flatpak/%s/desktop/%s.desktop/%s'.format(remote,
+                                                                    appId,
+                                                                    defaultBranch);
+    }
+
+    return appId;
+}
+
+function installAppFromStore(remote, appId) {
+    let appStoreId = getAppStoreAppId(remote, appId);
+    spawnProcess(['gnome-software', '--details=%s'.format(appStoreId)]);
```

Comment:
There are advantages with having g-s being spawned by dbus. Even if it's more code, since that code already exists, I'd prefer if we used it.